src/Simulator/IncrementalRedstoneSimulator/SolidBlockHandler.h

```diff
@@ -0,0 +1,71 @@
+
+#pragma once
+
+#include "RedstoneHandler.h"
+
+
+
+
+
+class cSolidBlockHandler : public cRedstoneHandler
+{
+	typedef cRedstoneHandler super;
+public:
+
+	cSolidBlockHandler(cWorld & a_World) :
+		super(a_World)
+	{
+	}
+
+	virtual unsigned char GetPowerDeliveredToPosition(const Vector3i & a_Position, BLOCKTYPE a_BlockType, NIBBLETYPE a_Meta, const Vector3i & a_QueryPosition, BLOCKTYPE a_QueryBlockType) override
+	{
+		// TODO: wire isn't linked powered only if the source was a wire, not just because it is a wire
+		return (
+			!cIncrementalRedstoneSimulator::IsRedstone(a_QueryBlockType) ||
+			(
+				(a_QueryBlockType == E_BLOCK_REDSTONE_WIRE) &&
+				(static_cast<cIncrementalRedstoneSimulator *>(m_World.GetRedstoneSimulator())->GetChunkData()->GetCachedPowerData(a_Position).PoweringBlock == E_BLOCK_REDSTONE_WIRE)
```

Comment:
See if you can get rid of some of these calls by passing in the cached power data. At some point in the chain, this is hiting cold data, and spending ages waiting on retrieving it from RAM.